### PR TITLE
feat(charts): allow query mutator to update queries after splitting original sql

### PR DESCRIPTION
**Pull request by solanksh**
feat(charts): allow query mutator to update queries after splitting original sql

### SUMMARY
The changes allow a user to use the SQL_QUERY_MUTATOR function to add statements such as SET ROLE without breaking functionality. Currently, adding a statemen like this works fine in the SQL Editor, but Charts run individual statements by splitting a passed query and throw errors since SET ROLE does not return any data on its own. With the new MUTATE_AFTER_SPLIT variable, users can choose to apply the mutator function after the chart's query execution function splits a given query, allowing users to apply a change to all statements in a query and have Chart functionality not be interrupted.

### TESTING INSTRUCTIONS
In the config file, update the SQL_QUERY_MUTATOR function to add a statement which does not return any results on its own and is something which should be run alongside all other statements in a given query. An example of this is adding a SET ROLE statement to the beginning of a query. Create a new Chart and see if queries return errors due to no data being fetched. Now in the config file, add a MUTATE_AFTER_SPLIT variable and set it equal to True. Create a new Chart and make a query, and verify that the desired results are showing.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [X ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/models/core.py [hunk 1]
```diff
@@ -415,7 +415,11 @@ def get_df(  # pylint: disable=too-many-locals
         mutator: Optional[Callable[[pd.DataFrame], None]] = None,
     ) -> pd.DataFrame:
         sqls = self.db_engine_spec.parse_sql(sql)
+
         engine = self.get_sqla_engine(schema)
+        username = utils.get_username() or username
```
Comment (baz-bakerhughes) [inline superset/models/core.py:420]: @villebro Comment from original Pull Request
This variable doesn't appear to be defined in this scope?

Comment (solanksh) [reply superset/models/core.py]: @villebro @baz-bakerhughes I have pushed the changes suggested.

**Comment by tooptoop4** [comment]
boomp

**Review by villebro — APPROVED** [review]
LGTM

**Comment by tooptoop4** [comment]
@solanksh this worked great as a patch on 2.0.1 but in 2.1.0 this no longer seems to work. after clicking 'create chart' on results from sqllab it then shows "Unexpected error
Error: no results to fetch"...when i 'view query' in the explore page it shows my additional set role query in front of the main query

**Comment by eschutho** [comment]
@tooptoop4 is the functionality only broken when the new config is set to `True`? I tested this with the flag off and couldn't reproduce.

**Comment by okayhooni** [comment]
@tooptoop4 

I got the same issue as yours..

Could you give me some some alternative method to bypass this issue..?

**Comment by tooptoop4** [comment]
did u solve @okayhooni ?

**Pull request merged**